Dappkit Redesign

medhakothari opened the pull request:
### Description

Uses styles similar to the rest of Valora, follows the designs [here](https://www.figma.com/file/zt7aTQ5wuXycIwxq5oAmF9/Wallet-Refresh?node-id=7424%3A468)

### Other changes

n/a

### Tested

##### Before:

<img src="https://user-images.githubusercontent.com/28763615/91213008-dc77b580-e6c5-11ea-9eb1-65566fd5cbd0.png" width ="200"/> <img src="https://user-images.githubusercontent.com/28763615/91213065-eac5d180-e6c5-11ea-9cff-2939dad68239.png" width ="200"/><img src="https://user-images.githubusercontent.com/28763615/91213101-f7e2c080-e6c5-11ea-8422-129ff70910b1.png" width ="200"/>

##### After:

<img src="https://user-images.githubusercontent.com/28763615/91213017-deda0f80-e6c5-11ea-8431-2da32abd6a4e.png" width ="200"/> <img src="https://user-images.githubusercontent.com/28763615/91213088-f44f3980-e6c5-11ea-9f6b-6085c3f8565b.png" width ="200"/><img src="https://user-images.githubusercontent.com/28763615/91213112-fb764780-e6c5-11ea-95ba-6f9af6a68dba.png" width ="200"/>


### Related issues

- Fixes #4644 

### Backwards compatibility

none

Commit:
made changes to account screen

Commit:
changed sign txn screen

Commit:
fixed data screen

medhakothari committed:
Merge branch 'master' into dappkit_design

Commit:
added spanish translations

medhakothari committed:
Merge branch 'master' into dappkit_design

medhakothari committed:
Update global.json

medhakothari committed:
Update global.json

jeanregisser commented on packages/mobile/src/dappkit/DappKitAccountScreen.tsx:
```diff
@@ -119,42 +114,40 @@ const styles = StyleSheet.create({
     alignItems: 'center',
     justifyContent: 'center',
     marginHorizontal: '15%',
+    marginVertical: '50%',
```
I was surprised by this at first and did some more checks to be sure it works in all situations, 
turns out it starts clipping the content on smaller screens, like iPhone SE:

<img src="https://user-images.githubusercontent.com/57791/91580836-f5929780-e94d-11ea-9145-bad229045cc4.png" width="50%" />

Here's the full styles that make it work on all screen sizes, no matter what the content is, the trick is with `flexGrow: 1` and fixing the button taking the full height:

```jsx
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  scrollContainer: {
    flexGrow: 1,
    alignItems: 'center',
    justifyContent: 'center',
    marginHorizontal: '15%',
  },
  header: {
    ...fontStyles.h1,
    alignItems: 'center',
    paddingBottom: 16,
  },
  share: {
    ...fontStyles.subSmall,
    color: colors.gray4,
    fontSize: 16,
    textAlign: 'center',
  },
  sectionDivider: {
    alignItems: 'center',
    width: 200,
  },
  sectionHeaderText: {
    ...fontStyles.sectionLabelNew,
    fontSize: 13,
    marginTop: 16,
  },
  bodyText: {
    ...fontStyles.regular,
    fontSize: 16,
    color: colors.gray4,
    textAlign: 'center',
  },
  button: {
    marginTop: 24,
  },
})
```

Result on iPhone SE and iPhone 11:

<img src="https://user-images.githubusercontent.com/57791/91582246-d268e780-e94f-11ea-8300-7b44f67aa37f.png" width="50%" /><img src="https://user-images.githubusercontent.com/57791/91580577-9c2a6880-e94d-11ea-9b2a-0146f2a7f28c.png" width="50%" />

jeanregisser commented on packages/mobile/src/dappkit/DappKitAccountScreen.tsx:
```diff
@@ -1,4 +1,4 @@
-import Button, { BtnTypes } from '@celo/react-components/components/Button'
+import Button, { BtnSizes, BtnTypes } from '@celo/react-components/components/Button.v2'
 import colors from '@celo/react-components/styles/colors'
 import fontStyles from '@celo/react-components/styles/fonts'
```
Redesigned screens should use `fonts.v2`.

jeanregisser commented on packages/mobile/src/dappkit/DappKitAccountScreen.tsx:
```diff
@@ -119,42 +114,40 @@ const styles = StyleSheet.create({
-    alignSelf: 'center',
+    ...fontStyles.subSmall,
+    color: colors.gray4,
+    fontSize: 16,
```
Once this is switched to `fonts.v2`, we should avoid redefining `fontSize`, unless absolutely unique and necessary.

jeanregisser commented on packages/mobile/src/dappkit/DappKitSignTxScreen.tsx:
```diff
@@ -1,4 +1,4 @@
-import Button, { BtnTypes } from '@celo/react-components/components/Button'
+import Button, { BtnSizes, BtnTypes } from '@celo/react-components/components/Button.v2'
 import colors from '@celo/react-components/styles/colors'
 import fontStyles from '@celo/react-components/styles/fonts'
```
ditto about using `font.v2`

jeanregisser commented on packages/mobile/src/dappkit/DappKitSignTxScreen.tsx:
```diff
@@ -121,24 +117,18 @@ const styles = StyleSheet.create({
     alignItems: 'center',
     justifyContent: 'center',
     marginHorizontal: '15%',
+    marginVertical: '50%',
```
ditto about using `flexGrow: 1` to prevent content clipping on smaller screen sizes.

jeanregisser commented on packages/mobile/src/dappkit/DappKitSignTxScreen.tsx:
```diff
@@ -147,22 +137,25 @@ const styles = StyleSheet.create({
+  button: {
+    flex: 1,
+    flexDirection: 'column',
+    alignItems: 'center',
```
These can be removed when using `flexGrow: 1` in the `contentContainerStyle`.

jeanregisser reviewed (changes requested):
Thanks!! 👍 
Found an issue on smaller screen sizes and added recommendations to simplify font styles following the redesign.

Commit:
changed to fonts v2

medhakothari commented:
Thanks for the helpful comments @jeanregisser! Didn't know fonts v2 existed, switched to that now! Also changed the button styling and tested on smaller screens, works now :)

<img width="354" alt="Screen Shot 2020-08-28 at 6 44 26 PM" src="https://user-images.githubusercontent.com/28763615/91625713-81142480-e95e-11ea-94d3-9d90adef8727.png">

jeanregisser commented on packages/mobile/src/dappkit/DappKitAccountScreen.tsx:
```diff
@@ -76,10 +76,13 @@ class DappKitAccountAuthScreen extends React.Component<Props> {
+        <TopBarTextButton
+          title={t('cancel')}
+          onPress={this.cancel}
+          titleStyle={{ color: colors.dark }}
```
nit: We try to avoid declaring styles inline with the component, unless necessary (animation or dynamic style).

> - By moving styles away from the render function, you're making the code easier to understand.
> - Naming the styles is a good way to add meaning to the low level components in the render function.

Source: https://reactnative.dev/docs/stylesheet.html

Another reason is about performance (though not the case here).
If the underlying component uses `PureComponent / React.memo`, passing a new style like this will defeat the render optimization when the parent component re-render.

jeanregisser commented on packages/mobile/src/dappkit/DappKitSignTxScreen.tsx:
```diff
@@ -74,10 +74,13 @@ class DappKitSignTxScreen extends React.Component<Props> {
+        <TopBarTextButton
+          title={t('cancel')}
+          onPress={this.cancel}
+          titleStyle={{ color: colors.dark }}
```
ditto about the inline style

jeanregisser reviewed (approved):
Looks great! 👍 
Left one optional comment.

annakaz reviewed (approved):
🚀

LGTM overall! +1 to Jean's comment on avoiding inline styles

Commit:
no inline button styling

medhakothari committed:
Merge branch 'master' into dappkit_design

medhakothari committed:
Merge branch 'master' into dappkit_design

medhakothari committed:
Merge branch 'master' into dappkit_design

medhakothari committed:
Merge branch 'master' into dappkit_design

medhakothari committed:
Merge branch 'master' into dappkit_design

Lss-Ankit commented:
@jeanregisser  Can you please provide a build details to verify task. thanks!
